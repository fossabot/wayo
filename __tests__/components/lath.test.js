import 'babel-polyfill';
import { shallow,createLocalVue,mount } from 'vue-test-utils';
import { render } from '@vue/server-test-utils';
import Lath from '../../src/components/lath';

const LocalVue = createLocalVue();

describe('Logic Cases', () => {
  const Wrapper = mount(Lath, {
    localVue: LocalVue
  });

  test('border', () => {
    expect(Wrapper.contains('.wayo-lath__separator-top')).toBe(false);
    expect(Wrapper.contains('.wayo-lath__separator-bottom')).toBe(false);

    Wrapper.setProps({
      borderTop: true,
      borderBottom: true
    });
    expect(Wrapper.contains('.wayo-lath__separator-top')).toBe(true);
    expect(Wrapper.contains('.wayo-lath__separator-bottom')).toBe(true);
  });

  test('inline', () => {
    expect(Wrapper.classes()).not.toContain('wayo-lath_inline');
    expect(Wrapper.classes()).not.toContain('wayo-lath_icon');
    expect(Wrapper.classes()).not.toContain('wayo-lath_content_header');
    Wrapper.setProps({
      inline: true
    });
    expect(Wrapper.classes()).toContain('wayo-lath_inline');
  });

  test('icon', () => {
    expect(Wrapper.contains('.wayo-lath__icon-head')).toBe(false);

    Wrapper.setProps({
      icon: 'bus',
      iconPosition: 'top'
    });
    expect(Wrapper.contains('.wayo-lath__icon-head')).toBe(true);
    expect(Wrapper.contains('.wayo-lath__icon-head_top')).toBe(true);
  });

  test('styles', () => {
    expect(Wrapper.vm.styles).toBe('');
    
    Wrapper.setData({
      tail: {
        icon: 'bus'
      }
    });
    expect(Wrapper.vm.styles).toMatch(/padding\-right:35px;/);

    Wrapper.setData({
      tail: {
        icon: 'bus',
        label: 'tail'
      }
    });
    expect(Wrapper.vm.styles).toMatch(/padding\-right:95px;/);
  });
});

describe('Snapshot Cases', () => {
  const Wrapper = render(Lath, {
    localVue: LocalVue
  });

  test('',() => {
    expect(Wrapper).toMatchSnapshot();
  });
});