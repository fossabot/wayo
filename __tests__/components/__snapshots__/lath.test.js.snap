// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Cases of Lath  1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "wayo-lath",
      "data-server-rendered": "true",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Snapshot Cases of LathContent  1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "wayo-lath__content",
      "data-server-rendered": "true",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "wayo-lath__content-entity wayo-lath__content_single-line",
        },
        "children": Array [
          Object {
            "data": "
      content
    ",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "p",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "data": " ",
          "next": Object {
            "data": "",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "comment",
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "text",
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "data": " ",
        "next": Object {
          "data": "",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "comment",
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "wayo-lath__content-entity wayo-lath__content_single-line",
          },
          "children": Array [
            Object {
              "data": "
      content
    ",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "text",
      },
      Object {
        "data": "",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "data": " ",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "wayo-lath__content-entity wayo-lath__content_single-line",
            },
            "children": Array [
              Object {
                "data": "
      content
    ",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "text",
        },
        "type": "comment",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Snapshot Cases of LathHeader  1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "wayo-lath__head",
      "data-server-rendered": "true",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "wayo-lath__title",
        },
        "children": Array [
          Object {
            "data": "title",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "data": " ",
          "next": Object {
            "attribs": Object {
              "class": "wayo-lath__subtitle",
              "style": "margin-top:5px;",
            },
            "children": Array [
              Object {
                "data": "subtitle",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "text",
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "data": " ",
        "next": Object {
          "attribs": Object {
            "class": "wayo-lath__subtitle",
            "style": "margin-top:5px;",
          },
          "children": Array [
            Object {
              "data": "subtitle",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "wayo-lath__title",
          },
          "children": Array [
            Object {
              "data": "title",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "text",
      },
      Object {
        "attribs": Object {
          "class": "wayo-lath__subtitle",
          "style": "margin-top:5px;",
        },
        "children": Array [
          Object {
            "data": "subtitle",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "data": " ",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "wayo-lath__title",
            },
            "children": Array [
              Object {
                "data": "title",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "text",
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Snapshot Cases of LathTail  1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "wayo-lath__tail",
      "data-server-rendered": "true",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "wayo-lath__tail-label",
        },
        "children": Array [
          Object {
            "data": "tail",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "data": " ",
          "next": Object {
            "attribs": Object {
              "class": "wayo-lath__tail-icon",
              "color": "",
              "name": "bus",
            },
            "children": Array [],
            "name": "wayo-icon",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "color": undefined,
              "name": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "color": undefined,
              "name": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "text",
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "data": " ",
        "next": Object {
          "attribs": Object {
            "class": "wayo-lath__tail-icon",
            "color": "",
            "name": "bus",
          },
          "children": Array [],
          "name": "wayo-icon",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "color": undefined,
            "name": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "color": undefined,
            "name": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "wayo-lath__tail-label",
          },
          "children": Array [
            Object {
              "data": "tail",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "text",
      },
      Object {
        "attribs": Object {
          "class": "wayo-lath__tail-icon",
          "color": "",
          "name": "bus",
        },
        "children": Array [],
        "name": "wayo-icon",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "data": " ",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "wayo-lath__tail-label",
            },
            "children": Array [
              Object {
                "data": "tail",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "text",
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "color": undefined,
          "name": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "color": undefined,
          "name": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
